connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogTableSpec.scala

```diff
@@ -0,0 +1,343 @@
+    createDefaultKs()
+    spark.sql(
+      s"""CREATE TABLE $defaultKs.$testTable (
+         |key_1 Int, key_2 Int, key_3 Int,
+         |cc1 STRING, cc2 String, cc3 String,
+         |value String) USING cassandra
+         |PARTITIONED BY (key_1, key_2, key_3)
+         |TBLPROPERTIES (clustering_key='cc1.asc, cc2.desc, cc3.asc')""".stripMargin)
+    val rows = spark.sql(s"SHOW TBLPROPERTIES $defaultKs.$testTable").collect()
+
+    val results = rows.map( row => (row.getString(0), CassandraSourceUtil.parseProperty(row.getString(1)))).toMap
+    results("clustering_key").asInstanceOf[String] should include ("cc1.ASC,cc2.DESC,cc3.ASC")
+    results("gc_grace_seconds").asInstanceOf[String] should be("864000")
+  }
+
+  it should "alter table properties" in {
+    createDefaultKs()
+    spark.sql(
+      s"""CREATE TABLE $defaultKs.$testTable (
+         |key_1 Int, key_2 Int, key_3 Int,
+         |cc1 STRING, cc2 String, cc3 String,
+         |value String) USING cassandra
+         |PARTITIONED BY (key_1, key_2, key_3)
+         |TBLPROPERTIES (clustering_key='cc1.asc, cc2.desc, cc3.asc')""".stripMargin)
+
+    spark.sql(
+      s"""ALTER TABLE $defaultKs.$testTable SET TBLPROPERTIES ('gc_grace_seconds' = '22')"""
+    )
+    val result = getTable(defaultKs, testTable)
+      .getOptions
+      .get(fromInternal("gc_grace_seconds"))
+      .asInstanceOf[Int].toString
+
+    result should be("22")
+  }
+
+  it should "alter map table properties" in {
+    createDefaultKs()
+    spark.sql(
+      s"""CREATE TABLE $defaultKs.$testTable (
+         |key_1 Int, key_2 Int, key_3 Int,
+         |cc1 STRING, cc2 String, cc3 String,
+         |value String) USING cassandra
+         |PARTITIONED BY (key_1, key_2, key_3)
+         |TBLPROPERTIES (clustering_key='cc1.asc, cc2.desc, cc3.asc')""".stripMargin)
+
+    spark.sql(
+      s"""ALTER TABLE $defaultKs.$testTable SET TBLPROPERTIES (
+         |'compaction'='{class=SizeTieredCompactionStrategy,bucket_high=1001}'
+         |)""".stripMargin
+    )
+
+    val compactionOptions = getTable(defaultKs, testTable)
+      .getOptions.asScala
+      .get(fromInternal("compaction"))
+      .get
+      .asInstanceOf[java.util.Map[String, AnyRef]]
+      .asScala
+
+    compactionOptions.get("bucket_high").get.toString should be ("1001")
+  }
+
+  it should "alter add columns" in {
+    createDefaultKs()
+    spark.sql(
+      s"""CREATE TABLE $defaultKs.$testTable (
+         |key_1 Int, key_2 Int, key_3 Int,
+         |cc1 STRING, cc2 String, cc3 String,
+         |value String) USING cassandra
+         |PARTITIONED BY (key_1, key_2, key_3)
+         |TBLPROPERTIES (clustering_key='cc1.asc, cc2.desc, cc3.asc')""".stripMargin)
+
+    spark.sql(
+      s"""ALTER TABLE $defaultKs.$testTable ADD COLUMNS (newCol INT) """.stripMargin
+    )
+
+    getTable(defaultKs, testTable)
+      .getColumn(fromInternal("newCol"))
+      .get().getName.asInternal() shouldBe ("newCol")
```

Comment:
Worth confirming that _all_ columns (including the new one) are present now?

Reply:
Definitely